### PR TITLE
mybatis files

### PR DESCRIPTION
**Pull request by SuperWebMonkey**

### CODE REVIEW

File: src/main/java/db/dao/IFlightDAO.java [hunk 1]
```diff
@@ -3,5 +3,5 @@
 import db.models.Flight;
 
 public interface IFlightDAO extends IBaseDAO<Flight> {
-    Flight getFlightPrice(double db_price);
+    Flight getFlightByPrice(double db_price);
```
Comment (ankochergin) [inline src/main/java/db/dao/IFlightDAO.java]:
```suggestion
    Flight getFlightByPrice(double price);
```

File: src/main/java/db/dao/IOrderDAO.java [hunk 2]
```diff
@@ -3,5 +3,5 @@
 import db.models.Order;
 
 public interface IOrderDAO extends IBaseDAO<Order> {
-    Order getOrderByPayment(int db_payment);
+    Order getOrderByPayment(double db_payment);
```
Comment (ankochergin) [inline src/main/java/db/dao/IOrderDAO.java]:
```suggestion
    Order getOrderByPayment(double payment);
```

**Review by ankochergin — CHANGES_REQUESTED** [review]
please review whole code and make sure all your variables, methods and classes have correct naming

File: src/main/java/db/patterns/CustomerBuilder.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public CustomerBuilder lastName(String lastName) {
+        this.lastName = lastName;
+        return this;
+    }
+
+    public CustomerBuilder phone(String phone) {
+        this.phone = phone;
+        return this;
+    }
+
```
Comment (ankochergin) [inline src/main/java/db/patterns/CustomerBuilder.java]: rename methods to **set**CustomerId or **with**CustomerId
method name always should show action

File: src/main/java/db/patterns/DatabaseFactory.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    private static final Logger LOGGER = LogManager.getLogger(MyBatisSqlFactory.class);
+
+    public IDatabaseType getInstance(String type) {
+        if (type.equals("jdbc"))
+            return ConnectionPool.getInstance();
+        else if (type.equals("mybatis"))
+            return new MyBatisSqlFactory();
+        else
+            LOGGER.info("Invalid option. No database selected.");
+        return null;
```
Comment (ankochergin) [inline src/main/java/db/patterns/DatabaseFactory.java]: do not return null

File: src/main/java/db/patterns/DatabaseFactory.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import db.connectionpool.ConnectionPool;
+import db.utils.MyBatisSqlFactory;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class DatabaseFactory {
+    private static final Logger LOGGER = LogManager.getLogger(MyBatisSqlFactory.class);
+
+    public IDatabaseType getInstance(String type) {
+        if (type.equals("jdbc"))
```
Comment (ankochergin) [inline src/main/java/db/patterns/DatabaseFactory.java]: 1. using of `equals` here is not null-safe. It'll throw NullPointerException if type is null. you should always use `equals` method on variable which is 100% not null. In this case `"jdbc".equals(type)`

Comment (ankochergin) [reply src/main/java/db/patterns/DatabaseFactory.java]: 2. I think types of the database can be moved to enum

Comment (ankochergin) [reply src/main/java/db/patterns/DatabaseFactory.java]: 3. You return 2 different types of objects, I would suggest to have DAO factories for both jdbc and mybatis and get proper DAO factory by enum

**Review by ankochergin — CHANGES_REQUESTED** [review]

File: src/main/java/db/patterns/JDBCDAOFactory.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            case "Payment":
+                return new PaymentDAO();
+            case "PaymentType":
+                return new PaymentTypeDAO();
+            case "Staff":
+                return new StaffDAO();
+            case "Tour":
+                return new TourDAO();
+            default:
+                throw new UnknownDatabaseTypeException("Unknown database type exception.");
```
Comment (ankochergin) [inline src/main/java/db/patterns/JDBCDAOFactory.java:36]: I think proper message here should be "Unknown model provided"

File: src/main/java/db/patterns/MyBatisDAOFactory.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+            case "DrivingCompany":
+                return new DrivingCompanyDAO();
+            case "Flight":
+                return new FlightDAO();
+            case "Hotel":
+                return new HotelDAO();
+            case "Order":
+                return new OrderDAO();
+            default:
+                throw new UnknownDatabaseTypeException("Unknown database type exception.");
```
Comment (ankochergin) [inline src/main/java/db/patterns/MyBatisDAOFactory.java:27]: same here. This factory is creating dao, not connections

Comment (SuperWebMonkey) [reply src/main/java/db/patterns/MyBatisDAOFactory.java:27]: What data type should be returned if it is not the DAO class?

**Pull request merged**